frappe/model/delete_doc.py

```diff
@@ -238,6 +238,11 @@ def check_permission_and_not_submitted(doc):
 		)
 
 
+IGNORE_LINK_FIELDS = {
+	"amended_from",  # Amended documents are always cancelled
```

Comment:
This feels like small fix for the problem

![image](https://github.com/frappe/frappe/assets/9079960/7f9b6ebf-fe58-4341-9eac-39816881fd34)